src/System.Data.Common/tests/System.Data.Common.Tests.csproj

```diff
@@ -74,7 +77,9 @@
     <Compile Include="System\Data\InRowChangingEventExceptionTest.cs" />
     <Compile Include="System\Data\InvalidConstraintExceptionTest.cs" />
     <Compile Include="System\Data\MissingPrimaryKeyExceptionTest.cs" />
-    <Compile Include="System\Data\MonkeyDataSet.cs" />
+    <Compile Include="System\Data\MonkeyDataSet.cs">
```

Comment:
Forgot one... 😢

Reply:
@FransBouma I wouldn't spend too much time removing `<SubType>Component</SubType>` 
What we typically do on the team is, we make changes in VS and allow it to pollute the CSProj and then we open the CSProj in a text editor and fix the XML to remove the component tag and then run the tests. 

Alternatively, the csproj can be manually modified in a text editor to add the `Compile` tags and the changes are committed. After that work on the project in VS and after editing code, discard any unnecessary changes to Csproj after previous commit to make sure that it is clean. 

The reason I mention this cleanup is because VS tries to open Component marked files in Designer and many a times it fails and the developer has to right click on the file and select View Code. 

However this is definitely not the crux of the PR :) Let's push to keep the ball rolling and address the other discussion comments opened by @divega 
I could clean up the Component tags if they become a problem.

Reply:
@saurabh500 :) The issue with opening the file in the designer view is indeed annoying :) It can be overcome with the attribute I mentioned earlier: decorating a class with `[System.ComponentModel.DesignerCategory("Code")]` and then opening it in vs will open it in the code editor. I think the only problem is if you decorate a class with that, it obviously will never open in the designer ;) which is a problem for typed datasets and the like ;) 

But agreed let's finish the last points and then wrap it up :) I think we're almost done with the changes, only the point regarding the null check in the ProviderRegistration ctor is left I think?

Reply:
Yeah that seems like the only open discussion point right now.

Reply:
Didn’t know about `[System.ComponentModel.DesignerCategory("Code")]`. Thanks! I will leave it to @saurabh500 here but I will definitely consider using that in other repos. Even if it is a bit more code, it seems a small price to pay for not having to fight VS ever again on this.

@bricelam FYI